.github/workflows/autoformat.yml

```diff
@@ -145,20 +144,59 @@ jobs:
         run: |
           ./build-rdoc.sh
       # ************************************************************************
-      # Push changes
+      # Upload patch
       # ************************************************************************
-      - name: Push changes
+      - name: Create patch
+        id: patch
         run: |
-          if [ -n "$(git status --porcelain)" ]; then
-            git add -A
-            git commit -s -m "Autoformat: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
-            git push
-          fi
+          git diff > ${{ github.run_id }}.diff
+          reformatted=$([[ -s ${{ github.run_id }}.diff ]] && echo "true" || echo "false")
+          echo "reformatted=$reformatted" >> $GITHUB_OUTPUT
+
+      - name: Upload patch
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ github.run_id }}.diff
+          path: ${{ github.run_id }}.diff
+
+  push:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    needs: [check-comment, format]
+    if: ${{ needs.format.outputs.reformatted == 'true' }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ needs.check-comment.outputs.repository }}
+          ref: ${{ needs.check-comment.outputs.head_ref }}
+          # Set fetch-depth to merge the base branch
+          fetch-depth: 300
+          token: ${{ secrets.MLFLOW_AUTOMATION_TOKEN }}
```

Comment:
I thought about using `GITHUB_TOKEN`, but a commit pushed by `GITHUB_TOKEN` doesn't trigger GitHub Actions workflows, which is inconvenient.

Reply:
This job should be safer than the format job because we just run `git` commands.